pkg/otlp/model/translator/consumer.go

```diff
@@ -16,20 +16,51 @@ package translator
 
 import (
 	"context"
+	"encoding"
+	"fmt"
 
 	"github.com/DataDog/datadog-agent/pkg/quantile"
 )
 
 // MetricDataType is a timeseries-style metric type.
 type MetricDataType int
 
+var _ encoding.TextUnmarshaler = (*MetricDataType)(nil)
+var _ encoding.TextMarshaler = (MetricDataType)(Gauge)
```

Comment:
I had to spend some time to wrap my head around why this is here. Interesting. Are you checking both that: `Gauge` is of the `MetricDataType` type and that it implements the interface? Is the type check/cast really necessary?

Reply:
No I think it's fine.